src/purchase-bill/purchase-bill.controller.ts

```diff
@@ -0,0 +1,56 @@
+import { Body, Controller, Delete, Get, Param, Patch, Post } from '@nestjs/common';
+import { ApiOkResponse, ApiTags } from '@nestjs/swagger';
+import { CreatePurchaseBillDto } from './dto/create-purchase-bill.dto';
+import { UpdatePurchaseBillDto } from './dto/update-purchase-bill.dto';
+import { PurchaseBill } from './entities/purchase-bill.entity';
+import { PurchaseBillService } from './purchase-bill.service';
+
+@ApiTags('Purchase Bill')
+@Controller('purchase-bill')
+export class PurchaseBillController {
+  constructor(private readonly purchaseBillService: PurchaseBillService) { }
+
+  @ApiOkResponse({
+    type: PurchaseBill,
+    description: 'create a purchase bill',
+  })
+  @Post()
+  create(@Body() createPurchaseBillDto: CreatePurchaseBillDto) {
+    const id = createPurchaseBillDto.id;
+    return this.purchaseBillService.create(createPurchaseBillDto, id);
+  }
```

Comment:
i can not see a reason why are you taking the id out of create dto. do whatever you want to with the dto when you are at your repository layer

Reply:
Issue with naming. It works fine with id name and throws error on vendor_id